lib/derivativo/conversion/office_helpers.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+module Derivativo
+  module Conversion
+    module OfficeHelpers
+      PDF_HIGHER_COMPRESSION_CONVERSION_THRESHOLD = 30.megabytes
+
+      # Converts an input office output audiovisual file
+      def self.office_convert_to_pdf(src_file_path:, dst_file_path:)
+        office_convert_to_pdf_impl(src_file_path: src_file_path, dst_file_path: dst_file_path, soffice_binary_path: soffice_binary_path_from_config_or_path)
+
+        # If the generated file size is less than or equal to our desired size, then we're done.
+        generated_file_size = File.size(dst_file_path)
+        return unless converted_file_size_merits_higher_compression_attempt?(generated_file_size)
+
+        # If we got here, then the generated file is larger than our desired size.
+        # We'll attempt one more conversion with higher compression.
+
+        office_convert_to_pdf_impl(
+          src_file_path: src_file_path,
+          dst_file_path: dst_file_path,
+          soffice_binary_path: soffice_binary_path_from_config_or_path,
+          compression_integer: compression_value_for_first_try_file_size(generated_file_size)
+        )
+      end
+
+      def self.soffice_binary_path_from_config_or_path
+        DERIVATIVO['soffice_binary_path'] || Derivativo::UserPathHelper.which('soffice')
+      end
+
+      def self.converted_file_size_merits_higher_compression_attempt?(file_size)
+        # TODO: At some point, maybe change PDF_HIGHER_COMPRESSION_CONVERSION_THRESHOLD here into a configuration option instead
+        # of a hard-coded value, though that will also change the logic in the
+        # compression_value_for_first_try_file_size method.
+        file_size > PDF_HIGHER_COMPRESSION_CONVERSION_THRESHOLD
+      end
+
+      def self.compression_value_for_first_try_file_size(size)
+        return 8 if size > 2000.megabytes
```

Comment:
similarly to PDF_HIGHER_COMPRESSION_CONVERSION_THRESHOLD, create a "constant" for 2000.megabytes, assuming there is an appropriate name for what the value represents.

Reply:
I reviewed the formula for various values and it turns out that I can remove that line altogether.  Less code!  🎉